### PR TITLE
use build name to ensure that winrm password and other shared state i…

### PR DESCRIPTION
**Pull request by SwampDragons**
This should guarantee the files are unique for the shared state. cc @rickard-von-essen @mwhooker

### CODE REVIEW

File: helper/common/shared_state.go [hunk 1]
```diff
@@ -9,23 +9,23 @@ import (
...
-func sharedStateFilename(suffix string) string {
+func sharedStateFilename(suffix string, buildName string) string {
 	uuid := os.Getenv("PACKER_RUN_UUID")
-	return filepath.Join(os.TempDir(), fmt.Sprintf("packer-%s-%s", uuid, suffix))
+	return filepath.Join(os.TempDir(), fmt.Sprintf("packer-%s-%s-%s", uuid, suffix, buildName))
 }
 
-func SetSharedState(key string, value string) error {
-	return ioutil.WriteFile(sharedStateFilename(key), []byte(value), 0600)
+func SetSharedState(key string, value string, buildName string) error {
```
Comment (mwhooker) [inline helper/common/shared_state.go]: instead of changing the function signature, can we just modify the key before calling this?

**Comment by mwhooker** [comment]
for some reason I wasn't expecting this, but looks `PackerBuildName` is updated for every run of the provisioners?

File: builder/azure/arm/builder.go [hunk 2]
```diff
@@ -177,7 +177,8 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 			NewStepDeployTemplate(azureClient, ui, b.config, deploymentName, GetVirtualMachineDeployment),
 			NewStepGetIPAddress(azureClient, ui, endpointConnectType),
 			&StepSaveWinRMPassword{
-				Password: b.config.tmpAdminPassword,
+				Password:  b.config.tmpAdminPassword,
```
Comment (boumenot) [inline builder/azure/arm/builder.go]: Did the spacing rules get updated?  I am surprised this wasn't already spaced correctly.

File: helper/common/shared_state.go (hunk 1 above)
Comment (SwampDragons) [reply helper/common/shared_state.go]: We could, but that'd require future users of the function to know they need to have the build name in the signature.

**Comment by SwampDragons** [comment]
@mwhooker looks like, yeah. It's in the provisioner config.

File: builder/azure/arm/builder.go (hunk 2 above)
Comment (SwampDragons) [reply builder/azure/arm/builder.go]: it automatically re-spaces when new fields are added to the struct to make them line up.

Comment (boumenot) [reply builder/azure/arm/builder.go]: Now I see it.  Thank you!

File: helper/common/shared_state.go (hunk 1 above)
Comment (SwampDragons) [reply helper/common/shared_state.go]: I'd rather have it explicit, even though I have to change the function signature.

**Pull request merged**